tests/forte/processors/data_augment/algorithms/skeleton_op_test.py

```diff
@@ -0,0 +1,104 @@
+
+            # Testing the insert annotations method
+            new_text_1 = "There"
+            new_text_2 = "Avoid"
+            insert_attempt_1 = self.insert_annotations(
+                new_text_1, input_pack, 5
+            )
+            insert_attempt_2 = self.insert_annotations(
+                new_text_2, input_pack, 5
+            )
+
+            # Testing the delete annotations method
+            delete_attempt_1 = self.delete_annotations(annotation_list[0])
+            delete_attempt_2 = self.delete_annotations(annotation_list[0])
+
+            # Testing the replace annotations method
+            replaced_text = "Last"
+            replace_attempt_1 = self.replace_annotations(
+                annotation_list[-1], replaced_text
+            )
+            replace_attempt_2 = self.replace_annotations(
+                annotation_list[-1], replaced_text
+            )
+            return True
+        except:
+            return False
+
+
+class TestSkeletonOp(unittest.TestCase):
+    def setUp(self) -> None:
+        self.skeleton_op = DummyAugmenter(
+            {
+                "augment_entry": "ft.onto.base_ontology.Token",
+                "other_entry_policy": {}
+            }
+        )
+
+    def test_operations(self) -> None:
+
+        # Check if when initialized, is are the variables that hold
+        # the inserted, deleted and replaced annotations are empty.
+        insert_list = self.skeleton_op._inserted_annos_pos_len
+        delete_list = self.skeleton_op._deleted_annos_id
+        replace_list = self.skeleton_op._replaced_annos
+        self.assertFalse(insert_list)
+        self.assertFalse(delete_list)
+        self.assertFalse(replace_list)
+
+        # Testing the insert annotations method
+        data_pack = DataPack()
+        data_pack.set_text("Hello World Today")
+        token_1 = Token(data_pack, 0, 5)
+        token_2 = Token(data_pack, 6, 11)
+        token_3 = Token(data_pack, 12, 17)
+        data_pack.add_entry(token_1)
+        data_pack.add_entry(token_2)
+        data_pack.add_entry(token_3)
+
+        # Perform predefined augmentation and check if
+        # it worked correctly
+        augmented_data_pack = self.skeleton_op.perform_augmentation(data_pack)
+        self.assertEqual(augmented_data_pack.text, "There World Last")
+
+        insert_list = self.skeleton_op._inserted_annos_pos_len
+        delete_list = self.skeleton_op._deleted_annos_id
+        replace_list = self.skeleton_op._replaced_annos
+
+        self.assertEqual(list(insert_list[data_pack.pack_id].items()), [(5, 5)])
+
+        self.assertEqual(list(delete_list[data_pack.pack_id]), [token_1.tid])
+
+        expectations = [
+            (Span(token_1.begin, token_1.end), ""),
+            (Span(token_1.end, token_1.end), "There"),
+            (Span(token_3.begin, token_3.end), "Last"),
+        ]
+
+        self.assertEqual(list(replace_list[data_pack.pack_id]), expectations)
+
```

Comment:
The test cases didn't test for the exceptions.

Also, the code you adapted here that does all the deletion and insertion work, are coupled with a good set of test cases here: https://github.com/asyml/forte/blob/master/tests/forte/processors/base/data_augment_replacement_processor_test.py

What's your plan of keeping these test cases?

Reply:
This could be done when we integrate the Base Data Augment Processor to use the Skeleton Op

Reply:
The ones that needed the creation would take time. But for the rest, like `test_modify_index` and `test_multi_pack_copy_link_or_group`, I will be adding them to the next commit.